app/src/main/java/com/mmm/parq/activities/DriverActivity.java

```diff
@@ -614,6 +636,27 @@ private void redirectToLogin() {
         finish();
     }
 
+    private void setProfilePicture() {
+        final Activity activity = this;
+        Thread initializeUser = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    mUser = getUser().get();
+                    if (mUser == null || mFirebaseRef.getAuth() == null) {
+                        redirectToLogin();
+                        return;
+                    }
+                } catch (Exception e) {
+                    Log.e(TAG, "Error getting user: " + e.getMessage());
+                }
+
+                PictureUtils.setProfilePicture(activity, mUser, mProfileView);
```

Comment:
Any idea if there is a difference between doing what you did, setting a variable to capture the activity vs. using DriverActivity.this inside the inner function?